noir/aztec_macros/src/lib.rs

```diff
@@ -1275,7 +1275,11 @@ fn abstract_return_values(func: &NoirFunction) -> Option<Statement> {
                 }
                 UnresolvedTypeData::FieldElement => Some(make_return_push(expression.clone())),
                 _ => None,
+            };
+            if new_return.is_some() {
+                func.def_mut().body.0.remove(len - 1);
```

Comment:
Can we add a comment here explaining that what this removes is the original expression, and what would happen if we did not do it?

Reply:
You could also use pop, and use last() when you retrieve the last_statement, the intent would be clearer

Reply:
Done!